### PR TITLE
Fix phpstan phpdocs issues

### PR DESCRIPTION
**Pull request by adlacruzes**
Hi,

after reading #9450, I think we can adopt an intermediate step when increasing phpstan level.

Level 2 show ~280 errors and could be hard fixing them without a baseline. Instead of increase full level we can add explicit rules from a higher level and fix errors progressively easier.

This PR fixes phpdocs issues and add some phpstan level 2 rules to config file.

### CODE REVIEW

**Review by staabm — APPROVED** [review]
Lgtm

File: src/Composer/Repository/PlatformRepository.php [hunk 1]
```diff
@@ -599,7 +599,7 @@ private function addLibrary($name, $prettyVersion, $description = null, array $r
     /**
      * Check if a package name is a platform package.
      *
-     * @param $name
+     * @param mixed $name
```
Comment (stof) [inline src/Composer/Repository/PlatformRepository.php:602]: `string` actually

File: tests/Composer/Test/Util/AuthHelperTest.php [hunk 2]
```diff
@@ -514,8 +514,8 @@ public function testStoreAuthWithPromptInvalidAnswer()
     }
 
     /**
-     * @param $origin
-     * @param $auth
+     * @param mixed $origin
```
Comment (stof) [inline tests/Composer/Test/Util/AuthHelperTest.php:517]: `string`

File: tests/Composer/Test/Util/AuthHelperTest.php [hunk 3]
```diff
@@ -514,8 +514,8 @@ public function testStoreAuthWithPromptInvalidAnswer()
...
+     * @param mixed $auth
```
Comment (stof) [inline tests/Composer/Test/Util/AuthHelperTest.php:518]: `array`

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by stof** [comment]
> after reading #9450, I think we can adopt an intermediate step when increasing phpstan level.
> 
> Level 2 show ~280 errors and could be hard fixing them without a baseline. Instead of increase full level we can add explicit rules from a higher level and fix errors progressively easier.

I'd rather go to level 2 and introducing the baseline. that would be easier to maintain (no need to maintain a list of rules belonging to level 2) and would bring more value (new code would be checked at level 2).

the answer from @Seldaek on #9450 did not reject the usage of a baseline. This answer was rejecting the fact of adding a baseline without trying to fix as much reported issues as possible before generating it (so that moving to level 2 actually helps increasing the quality of the code).

File: src/Composer/Repository/PlatformRepository.php (hunk 1 above)
Comment (adlacruzes) [reply src/Composer/Repository/PlatformRepository.php]: Done

File: tests/Composer/Test/Util/AuthHelperTest.php (hunk 2 above)
Comment (adlacruzes) [reply tests/Composer/Test/Util/AuthHelperTest.php]: Done

File: tests/Composer/Test/Util/AuthHelperTest.php (hunk 3 above)
Comment (adlacruzes) [reply tests/Composer/Test/Util/AuthHelperTest.php]: Applied

**Comment by adlacruzes** [comment]
Hi @stof, I rechecked mixed types and applied your correct suggestions.

The idea is to remove these rules once level 2 is reached, but its true we are introducing one more thing to maintain.

Should I limit this pull request to the proposed fixes, so we can merge it?

Thanks!

**Comment by stof** [comment]
> The idea is to remove these rules once level 2 is reached, but its true we are introducing one more thing to maintain.

I'd rather do this by configuring level 2 with a baseline, and then work on emptying the baseline little by little to reach full level 2. this way, new code would be written at level 2 already.

**Comment by Seldaek** [comment]
Going to merge this as it already fixes some things and adds more strictness, it goes in the right direction. But yes full level 2 or more with baseline probably makes more sense in the long run.

**Comment by adlacruzes** [comment]
Thanks everyone!

**Pull request merged**